### PR TITLE
Add hive4 query runner and tests

### PR DESCRIPTION
**Pull request by mayankvadariya**
## Description

Add hive4 query runner and tests. Depends on https://github.com/trinodb/docker-images/pull/218

## Release notes

(x) This is not user-visible or is docs only, and no release notes are required.

```markdown
## Section
* Fix some things. ({issue}`issuenumber`)
```

### CODE REVIEW

File: plugin/trino-hive/src/test/java/io/trino/plugin/hive/containers/Hive4Metastore.java [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+
+public class Hive4Metastore
+        extends BaseTestContainer
+{
+    private static final Logger log = Logger.get(HiveHadoop.class);
+    private static final String HOST_NAME = "metastore";
+    public static final String HIVE4_IMAGE = "ghcr.io/trinodb/testing/hive4.0-hive:latest";
```
Comment (findinpath) [inline plugin/trino-hive/src/test/java/io/trino/plugin/hive/containers/Hive4Metastore.java:32]: You'd likely need to use `TestingProperties.getDockerImagesVersion()`

Comment (findinpath) [inline plugin/trino-hive/src/test/java/io/trino/plugin/hive/containers/Hive4Metastore.java:32]: Add to the description of the PR that it depends on https://github.com/trinodb/docker-images/pull/218

File: plugin/trino-hive/src/test/java/io/trino/plugin/hive/containers/Hive4MinioDataLake.java [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+                Minio.builder()
+                        .withNetwork(network)
+                        .withEnvVars(ImmutableMap.<String, String>builder()
+                                .put("MINIO_ACCESS_KEY", MINIO_ACCESS_KEY)
+                                .put("MINIO_SECRET_KEY", MINIO_SECRET_KEY)
+                                .put("MINIO_REGION", MINIO_DEFAULT_REGION)
+                                .buildOrThrow())
+                        .build());
+
+        Hive4Metastore.Builder metastorebuilder = Hive4Metastore.builder()
```
Comment (findinpath) [inline plugin/trino-hive/src/test/java/io/trino/plugin/hive/containers/Hive4MinioDataLake.java:79]: would be worth referencing the hive readme (use the tag version) containing the docker commands used to start the hive metastore and hive server

File: plugin/trino-hive/src/test/java/io/trino/plugin/hive/TestHive4OnDataLake.java [hunk 3]
```diff
@@ -0,0 +1,2282 @@
...
+import static java.util.stream.Collectors.joining;
+import static java.util.stream.Collectors.toSet;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.junit.jupiter.api.TestInstance.Lifecycle.PER_CLASS;
+
+@Execution(ExecutionMode.SAME_THREAD) // Custom hive4 image doesn't support concurrent queries
+@TestInstance(PER_CLASS)
+public class TestHive4OnDataLake
+        extends TestHive3OnDataLake
```
Comment (findinpath) [inline plugin/trino-hive/src/test/java/io/trino/plugin/hive/TestHive4OnDataLake.java:79]: Maybe `TestHive4OnDataLake` & `TestHive3OnDataLake` should extend from  `BaseTestHiveOnDataLake`

If you go ahead with this change, make the renaming of as part of a separated commit to keep the code changes easy to read.

Comment (findinpath) [reply plugin/trino-hive/src/test/java/io/trino/plugin/hive/TestHive4OnDataLake.java:79]: `runOnHive` could be an abstract method delegating to:

- `hiveMinioDataLake.getHiveHadoop().runOnHive` on hive3
- `hiveMinioDataLake.getHiveServer().runOnHive` on hive4

File: plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/catalog/hms/TestTrinoHive4CatalogWithHiveMetastore.java [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+
+@TestInstance(PER_CLASS)
+@Execution(CONCURRENT)
+public class TestTrinoHive4CatalogWithHiveMetastore
+        extends BaseTrinoCatalogTest
+{
+    private static final String bucketName = "test-hive-catalog-with-hms-" + randomNameSuffix();
+
+    private final AutoCloseableCloser closer = AutoCloseableCloser.create();
+    // Use MinIO for storage, since HDFS is hard to get working in a unit test
```
Comment (findinpath) [inline plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/catalog/hms/TestTrinoHive4CatalogWithHiveMetastore.java:71]: I'd argue that the comment is not relevant (anymore) - HDFS is not necessarily the target storage Trino is built for.

File: plugin/trino-hive/src/test/java/io/trino/plugin/hive/containers/Hive4MinioDataLake.java (hunk 2 above)
Comment (mayankvadariya) [reply plugin/trino-hive/src/test/java/io/trino/plugin/hive/containers/Hive4MinioDataLake.java:79]: ya thats good idea

File: plugin/trino-hive/src/test/java/io/trino/plugin/hive/TestHive4OnDataLake.java (hunk 3 above)
Comment (mayankvadariya) [reply plugin/trino-hive/src/test/java/io/trino/plugin/hive/TestHive4OnDataLake.java]: ya created an abstract class BaseTestHiveOnDataLake  which is extended by TestHive4OnDataLake and TestHive3OnDataLake

File: plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/catalog/hms/TestTrinoHive4CatalogWithHiveMetastore.java (hunk 4 above)
Comment (mayankvadariya) [reply plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/catalog/hms/TestTrinoHive4CatalogWithHiveMetastore.java]: removed it

File: plugin/trino-hive/src/test/java/io/trino/plugin/hive/containers/Hive4MinioDataLake.java [hunk 5]
```diff
@@ -0,0 +1,179 @@
...
+
+public class Hive4MinioDataLake
+        implements AutoCloseable
+{
+    /**
+     * In S3 this region is implicitly the default one. In Minio, however,
+     * if we set an empty region, it will accept any.
+     * So setting it by default to `us-east-1` simulates S3 better
+     */
+    public static final String MINIO_DEFAULT_REGION = "us-east-1";
```
Comment (anusudarsan) [inline plugin/trino-hive/src/test/java/io/trino/plugin/hive/containers/Hive4MinioDataLake.java:42]: private

File: plugin/trino-hive/src/test/java/io/trino/plugin/hive/TestHive4OnDataLake.java [hunk 6]
```diff
@@ -0,0 +1,128 @@
...
+import io.trino.testing.minio.MinioClient;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.parallel.Execution;
+import org.junit.jupiter.api.parallel.ExecutionMode;
+
+import static io.trino.plugin.hive.TestingThriftHiveMetastoreBuilder.testingThriftHiveMetastoreBuilder;
+import static io.trino.testing.TestingNames.randomNameSuffix;
+import static org.junit.jupiter.api.Assumptions.abort;
+
+@Execution(ExecutionMode.SAME_THREAD) // Custom hive4 image doesn't support running concurrent queries
```
Comment (anusudarsan) [inline plugin/trino-hive/src/test/java/io/trino/plugin/hive/TestHive4OnDataLake.java:32]: add this a TODO

File: plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/catalog/hms/TestTrinoHive4CatalogWithHiveMetastore.java [hunk 7]
```diff
@@ -0,0 +1,145 @@
...
+
+    @AfterAll
+    public void tearDown()
+            throws Exception
+    {
+        dataLake = null;
+        closer.close();
+    }
+
+    @Override
```
Comment (anusudarsan) [inline plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/catalog/hms/TestTrinoHive4CatalogWithHiveMetastore.java:88]: can this simply extend `TestTrinoHiveCatalogWithHiveMetastore` ?

File: plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/catalog/hms/TestTrinoHive4CatalogWithHiveMetastore.java [hunk 8]
```diff
@@ -0,0 +1,145 @@
...
+import static io.trino.testing.TestingNames.randomNameSuffix;
+import static io.trino.testing.containers.Minio.MINIO_ACCESS_KEY;
+import static io.trino.testing.containers.Minio.MINIO_SECRET_KEY;
+import static java.util.concurrent.TimeUnit.MINUTES;
+import static org.junit.jupiter.api.TestInstance.Lifecycle.PER_CLASS;
+import static org.junit.jupiter.api.parallel.ExecutionMode.CONCURRENT;
+
+@TestInstance(PER_CLASS)
+@Execution(CONCURRENT)
+public class TestTrinoHive4CatalogWithHiveMetastore
```
Comment (anusudarsan) [inline plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/catalog/hms/TestTrinoHive4CatalogWithHiveMetastore.java:65]: just so you know, this is an iceberg catalog test. was that the intention?

File: plugin/trino-hive/src/test/java/io/trino/plugin/hive/TestHive4OnDataLake.java [hunk 9]
```diff
@@ -0,0 +1,128 @@
...
+    String runOnHive(String query)
+    {
+        return hiveMinioDataLake.getHiveServer().runOnHive(query);
+    }
+
+    @Override
+    @Test
+    public void testSyncPartitionOnBucketRoot()
+    {
+        abort("Fails with `location must not be root path`");
```
Comment (anusudarsan) [inline plugin/trino-hive/src/test/java/io/trino/plugin/hive/TestHive4OnDataLake.java:105]: maybe these should be linked to issues in github (for tracking purposes)

File: plugin/trino-hive/src/test/java/io/trino/plugin/hive/TestHive4OnDataLake.java [hunk 10]
```diff
@@ -0,0 +1,128 @@
...
+
+@Execution(ExecutionMode.SAME_THREAD) // Custom hive4 image doesn't support running concurrent queries
+class TestHive4OnDataLake
+        extends BaseTestHiveOnDataLake
+{
+    private String bucketName;
+    private Hive4MinioDataLake hiveMinioDataLake;
+    private HiveMetastore metastoreClient;
+
+    @Override
```
Comment (anusudarsan) [inline plugin/trino-hive/src/test/java/io/trino/plugin/hive/TestHive4OnDataLake.java:40]: move common code from createQueryRunner() as well to BaseTestHiveOnDataLake

File: plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/catalog/hms/TestTrinoHive4CatalogWithHiveMetastore.java (hunk 8 above)
Comment (mayankvadariya) [reply plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/catalog/hms/TestTrinoHive4CatalogWithHiveMetastore.java:65]: yes I picked up an Iceberg test which uses hms. I can remove it if its unnecessary.

File: plugin/trino-hive/src/test/java/io/trino/plugin/hive/BaseTestHiveOnDataLake.java
Comment (hashhar) [inline plugin/trino-hive/src/test/java/io/trino/plugin/hive/BaseTestHiveOnDataLake.java:1]: " Extract BaseTestHiveOnDataLake to reuse it across Hive3/4 test "

Ideally you should do this in two steps (rename only change + separate change to use the new base class in the concrete class). Preserves the history that way.

**Review by hashhar — COMMENTED** [review]
LGTM % Anu's comments.

File: plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/catalog/hms/TestTrinoHive4CatalogWithHiveMetastore.java (hunk 8 above)
Comment (anusudarsan) [reply plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/catalog/hms/TestTrinoHive4CatalogWithHiveMetastore.java:23]: no we can keep this. just fyi

**Comment by ebyhr** [comment]
https://github.com/trinodb/docker-images release is complete.

File: plugin/trino-hive/src/test/java/io/trino/plugin/hive/TestHive4OnDataLake.java (hunk 9 above)
Comment (mayankvadariya) [reply plugin/trino-hive/src/test/java/io/trino/plugin/hive/TestHive4OnDataLake.java]: Good thought. Linked issues instead now.

File: plugin/trino-hive/src/test/java/io/trino/plugin/hive/BaseTestHiveOnDataLake.java
Comment (mayankvadariya) [reply plugin/trino-hive/src/test/java/io/trino/plugin/hive/BaseTestHiveOnDataLake.java:1]: Good suggestion.

File: plugin/trino-hive/src/test/java/io/trino/plugin/hive/TestHive4OnDataLake.java (hunk 10 above)
Comment (mayankvadariya) [reply plugin/trino-hive/src/test/java/io/trino/plugin/hive/TestHive4OnDataLake.java]: done

File: plugin/trino-hive/src/test/java/io/trino/plugin/hive/TestHive3OnDataLake.java [hunk 11]
```diff
@@ -80,22 +80,22 @@ public class TestHive3OnDataLake
...
-    private HiveMinioDataLake hiveMinioDataLake;
+    private Hive3MinioDataLake hive3MinioDataLake;
     private HiveMetastore metastoreClient;
 
     @Override
     protected QueryRunner createQueryRunner()
             throws Exception
     {
         this.bucketName = "test-hive-insert-overwrite-" + randomNameSuffix();
-        this.hiveMinioDataLake = closeAfterClass(
```
Comment (anusudarsan) [inline plugin/trino-hive/src/test/java/io/trino/plugin/hive/TestHive3OnDataLake.java:91]: I think we can leave the private object name as-is hiveMinioDataLake (here and others). Reduces the diff, and also it looks like you are naming it all back when this is converted to an abstract class [later](https://github.com/trinodb/trino/pull/24418/commits/661fcec1974cfc6bba172b01ac83c87dce95eb2b#diff-c6ac76ce5a102f34fe5cdca7794cff7404bf8ed13740ae7232c25ad6205a1245).

File: plugin/trino-hive/src/test/java/io/trino/plugin/hive/TestHive4OnDataLake.java [hunk 12]
```diff
@@ -0,0 +1,61 @@
...
+    public void testSyncPartitionOnBucketRoot()
+    {
+        abort("https://github.com/trinodb/trino/issues/24453");
+    }
+
+    @Override
+    @Test
+    public void testUnpartitionedTableExternalLocationOnTopOfTheBucket()
+    {
+        abort("https://github.com/trinodb/trino/issues/24453");
```
Comment (anusudarsan) [inline plugin/trino-hive/src/test/java/io/trino/plugin/hive/TestHive4OnDataLake.java:45]: nit: I didnt mean that you have to change the abort message. More like adding a code comment with issue links

File: plugin/trino-hive/src/test/java/io/trino/plugin/hive/TestHive3OnDataLake.java (hunk 11 above)
Comment (mayankvadariya) [reply plugin/trino-hive/src/test/java/io/trino/plugin/hive/TestHive3OnDataLake.java:91]: I personally like to keep the name as that of the class but I'll revert as you suggested. ya I renamed it later for one of the class as in that commit `Hive3MinioDataLake` becomes `HiveMinioDataLake`

File: plugin/trino-hive/src/test/java/io/trino/plugin/hive/containers/HiveMinioDataLake.java [hunk 13]
```diff
@@ -0,0 +1,135 @@
...
+    private final Minio minio;
+    private MinioClient minioClient;
+
+    protected final AutoCloseableCloser closer = AutoCloseableCloser.create();
+    protected final Network network;
+    protected State state = INITIAL;
+
+    public HiveMinioDataLake(String bucketName)
+    {
+        this.bucketName = requireNonNull(bucketName, "bucketName is null");
```
[the remark this answers is not in the document]
Comment (mayankvadariya) [reply plugin/trino-hive/src/test/java/io/trino/plugin/hive/containers/HiveMinioDataLake.java:50]: I feel that bucket name is taken as an input to provide meaningful bucket name by caller as HiveMinioDataLake is used across classes.

**Comment by mayankvadariya** [comment]
@hashhar please review.

File: plugin/trino-hive/src/test/java/io/trino/plugin/hive/TestHive4OnDataLake.java (hunk 12 above)
Comment (mayankvadariya) [reply plugin/trino-hive/src/test/java/io/trino/plugin/hive/TestHive4OnDataLake.java]: sure added a code comment with issue link.

File: plugin/trino-hive/src/test/java/io/trino/plugin/hive/containers/Hive4Metastore.java [hunk 14]
```diff
@@ -0,0 +1,96 @@
...
+
+import static io.trino.testing.TestingProperties.getDockerImagesVersion;
+
+public class Hive4Metastore
+        extends BaseTestContainer
+{
+    private static final Logger log = Logger.get(HiveHadoop.class);
+    private static final String HOST_NAME = "metastore";
+
+    public static final String HIVE4_IMAGE = "ghcr.io/trinodb/testing/hive4.0-hive:" + getDockerImagesVersion();
```
Comment (anusudarsan) [inline plugin/trino-hive/src/test/java/io/trino/plugin/hive/containers/Hive4Metastore.java:35]: move public to top

File: plugin/trino-hive/src/test/java/io/trino/plugin/hive/containers/Hive4HiveServer.java [hunk 15]
```diff
@@ -0,0 +1,95 @@
...
+
+import static io.trino.plugin.hive.containers.Hive4Metastore.HIVE4_IMAGE;
+
+public class Hive4HiveServer
+        extends BaseTestContainer
+{
+    private static final Logger log = Logger.get(Hive4HiveServer.class);
+    private static final String HOST_NAME = "hiveserver2";
+
+    public static final int HIVE_SERVER_PORT = 10000;
```
Comment (anusudarsan) [inline plugin/trino-hive/src/test/java/io/trino/plugin/hive/containers/Hive4HiveServer.java:35]: move to top

File: plugin/trino-hive/src/test/java/io/trino/plugin/hive/containers/Hive4HiveServer.java [hunk 16]
```diff
@@ -0,0 +1,95 @@
...
+
+    public static Builder builder()
```
Comment (anusudarsan) [inline plugin/trino-hive/src/test/java/io/trino/plugin/hive/containers/Hive4HiveServer.java:37]: move the Builder class here

File: plugin/trino-hive/src/test/java/io/trino/plugin/hive/containers/Hive4Metastore.java [hunk 17]
```diff
@@ -0,0 +1,96 @@
...
+    public static final int HIVE_METASTORE_PORT = 9083;
+
+    public static Builder builder()
```
Comment (anusudarsan) [inline plugin/trino-hive/src/test/java/io/trino/plugin/hive/containers/Hive4Metastore.java:38]: keep the builder() closer to class

File: plugin/trino-hive/src/test/resources/hive_minio_datalake/hive4-hive-site.xml [hunk 18]
```diff
@@ -0,0 +1,78 @@
...
+        <value>local</value>
+    </property>
+    <property>
+        <name>tez.local.mode</name>
+        <value>true</value>
+    </property>
+    <property>
+        <name>hive.execution.engine</name>
+        <value>tez</value>
+    </property>
```
Comment (anusudarsan) [inline plugin/trino-hive/src/test/resources/hive_minio_datalake/hive4-hive-site.xml:39]: can you look at thees properties and see if these are required along with mapreduce  properties? I would assume hive requires one of tez or mapreduce properties. not both

File: plugin/trino-hive/src/test/java/io/trino/plugin/hive/containers/HiveMinioDataLake.java (hunk 13 above)
[the remark this answers is not in the document]
Comment (anusudarsan) [reply plugin/trino-hive/src/test/java/io/trino/plugin/hive/containers/HiveMinioDataLake.java:50]: > by caller as HiveMinioDataLake is used across classes.

ya but does not look like `Test... CatalogWithHiveMetastore` does the same..nbd

**Review by anusudarsan — COMMENTED** [review]
few nits. LGTM

File: plugin/trino-hive/src/test/resources/hive_minio_datalake/hive4-hive-site.xml (hunk 18 above)
Comment (mayankvadariya) [reply plugin/trino-hive/src/test/resources/hive_minio_datalake/hive4-hive-site.xml]: Good point. Tests passes without these properties as well. However, there might be some default fallbacks for these props. I removed them from this file.

File: plugin/trino-hive/src/test/java/io/trino/plugin/hive/BaseTestHiveOnDataLake.java [hunk 19]
```diff
@@ -73,27 +72,38 @@
...
 
-    private String bucketName;
-    private Hive3MinioDataLake hiveMinioDataLake;
-    private HiveMetastore metastoreClient;
+    protected String bucketName;
+    protected HiveMinioDataLake hiveMinioDataLake;
+    protected HiveMetastore metastoreClient;
+
+    @BeforeAll
+    public void setUp()
```
Comment (ebyhr) [inline plugin/trino-hive/src/test/java/io/trino/plugin/hive/BaseTestHiveOnDataLake.java:86]: Why did you move this method? The original place looks good to me. This setup should happen after `createQueryRunner()` method.

File: plugin/trino-hive/src/test/java/io/trino/plugin/hive/BaseTestHiveOnDataLake.java [hunk 20]
```diff
@@ -73,27 +72,38 @@
...
+    {
+        computeActual(format(
+                "CREATE SCHEMA hive.%1$s WITH (location='s3a://%2$s/%1$s')",
+                HIVE_TEST_SCHEMA,
+                bucketName));
+        computeActual("CREATE SCHEMA hive.functions");
+    }
+
+    protected abstract HiveMinioDataLake hiveMinioDataLake();
```
Comment (ebyhr) [inline plugin/trino-hive/src/test/java/io/trino/plugin/hive/BaseTestHiveOnDataLake.java:95]: We could add a constructor that accepts `HiveMinioDataLake` instead of adding a protected method.

File: plugin/trino-hive/src/test/java/io/trino/plugin/hive/s3/S3HiveQueryRunner.java [hunk 21]
```diff
@@ -187,4 +188,24 @@ public static void main(String[] args)
...
+            Hive4MinioDataLake hiveMinioDataLake = new Hive4MinioDataLake("tpch");
+            hiveMinioDataLake.start();
+
+            QueryRunner queryRunner = S3HiveQueryRunner.builder(hiveMinioDataLake)
+                    .addCoordinatorProperty("http-server.http.port", "8080")
+                    .setHiveProperties(ImmutableMap.of("hive.security", "allow-all"))
+                    .setSkipTimezoneSetup(true)
+                    .setInitialTables(TpchTable.getTables())
+                    .build();
+            Logger log = Logger.get(S3HiveQueryRunner.class);
```
Comment (ebyhr) [inline plugin/trino-hive/src/test/java/io/trino/plugin/hive/s3/S3HiveQueryRunner.java:206]: The class should be `S3Hive4QueryRunner`.

File: plugin/trino-hive/src/test/java/io/trino/plugin/hive/containers/Hive4MinioDataLake.java [hunk 22]
```diff
@@ -0,0 +1,99 @@
...
+    {
+        this(bucketName, Hive4Metastore.HIVE4_IMAGE);
+    }
+
+    public Hive4MinioDataLake(String bucketName, String hiveImage)
+    {
+        this(bucketName, ImmutableMap.of("/opt/hive/conf/hive-site.xml", getPathFromClassPathResource("hive_minio_datalake/hive4-hive-site.xml")), hiveImage);
+    }
+
+    public Hive4MinioDataLake(String bucketName, Map<String, String> hiveFilesToMount, String hiveImage)
```
Comment (ebyhr) [inline plugin/trino-hive/src/test/java/io/trino/plugin/hive/containers/Hive4MinioDataLake.java:44]: Please merge redundant constructors into one.

File: plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/catalog/hms/TestTrinoHive4CatalogWithHiveMetastore.java [hunk 23]
```diff
@@ -0,0 +1,27 @@
...
+package io.trino.plugin.iceberg.catalog.hms;
+
+import io.trino.plugin.hive.containers.Hive4MinioDataLake;
+import io.trino.plugin.hive.containers.HiveMinioDataLake;
+
+public class TestTrinoHive4CatalogWithHiveMetastore
+        extends TestTrinoHiveCatalogWithHiveMetastore
+{
+    @Override
+    HiveMinioDataLake hiveMinioDataLake()
```
Comment (ebyhr) [inline plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/catalog/hms/TestTrinoHive4CatalogWithHiveMetastore.java:23]: We could add a constructor instead.

**Review by ebyhr — APPROVED** [review]

File: plugin/trino-hive/src/test/java/io/trino/plugin/hive/BaseTestHiveOnDataLake.java (hunk 19 above)
Comment (mayankvadariya) [reply plugin/trino-hive/src/test/java/io/trino/plugin/hive/BaseTestHiveOnDataLake.java]: It was by mistake.

File: plugin/trino-hive/src/test/java/io/trino/plugin/hive/BaseTestHiveOnDataLake.java (hunk 20 above)
Comment (mayankvadariya) [reply plugin/trino-hive/src/test/java/io/trino/plugin/hive/BaseTestHiveOnDataLake.java]: added a constructor.

File: plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/catalog/hms/TestTrinoHive4CatalogWithHiveMetastore.java (hunk 23 above)
Comment (mayankvadariya) [reply plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/catalog/hms/TestTrinoHive4CatalogWithHiveMetastore.java:23]: This wasn't possible as it requires two constructors(with HiveMinioDataLake parameter and a default one). However, test framework doesn't seem to support multiple constructors.

File: plugin/trino-hive/src/test/java/io/trino/plugin/hive/BaseTestHiveOnDataLake.java (hunk 20 above)
Comment (mayankvadariya) [reply plugin/trino-hive/src/test/java/io/trino/plugin/hive/BaseTestHiveOnDataLake.java]: test failed with https://github.com/trinodb/trino/actions/runs/12402571060/job/34624391815?pr=24418#step:5:2540 which had failed with the same error before as well. I'll revert this to use abstract method.

Comment (ebyhr) [reply plugin/trino-hive/src/test/java/io/trino/plugin/hive/BaseTestHiveOnDataLake.java]: The error means we shouldn't reuse the same bucket name among tests. You can see a comment in `MinioClient#makeBucket`. The cause is you changed `bucketName` to static.

Comment (mayankvadariya) [reply plugin/trino-hive/src/test/java/io/trino/plugin/hive/BaseTestHiveOnDataLake.java]: ya I saw that but wasn't able to reproduce it. I think it could have been reproduced if I would have run in parallel.I moved bucket name to sub classes now.

**Comment by mayankvadariya** [comment]
https://github.com/trinodb/trino/actions/runs/12404758413/job/34630462973?pr=24418 looks like a unrelated failure.
